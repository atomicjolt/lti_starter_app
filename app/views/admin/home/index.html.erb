<% content_for :head do -%>
  <%= webpack_styles_tag "admin", "admin"%>
<% end -%>

<%= render 'shared/default_admin_client_settings' %>
<%= webpack_manifest_script "admin" %>
<div id="main-app"></div>
<%= webpack_bundle_tag "admin",  "admin_manifest"%>
<%= webpack_bundle_tag "admin",  "admin_vendor"%>
<%= webpack_bundle_tag "admin",  "admin"%>
