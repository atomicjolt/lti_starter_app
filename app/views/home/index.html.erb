<% content_for :head do -%>
  <%= stylesheet_link_tag "hello_world" %>
<% end -%>

<%= render "shared/default_client_settings" %>
<div id="main-app"></div>
<%= javascript_include_tag "hello_world", defer: true %>
