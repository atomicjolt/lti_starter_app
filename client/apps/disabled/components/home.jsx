import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import assets from '../libs/assets';
import Auth from '../../../libs/canvas/components/canvas_authentication';
import Selector from './content_item_select/selector';

const select = state => ({
  canvasAuthRequired: state.settings.canvas_auth_required,
  ltiMessageType: state.settings.lti_message_type,
});

const Home = (props) => {

  const img = assets('./images/atomicjolt.jpg');

  if (props.ltiMessageType === 'ContentItemSelectionRequest') {
    return <Selector />;
  }

  return (
    <div>
      <img src={img} alt="Atomic Jolt Logo" />
      <h1>This account has been disabled</h1>
    </div>
  );

};

Home.propTypes = {
  canvasAuthRequired: PropTypes.bool,
  ltiMessageType: PropTypes.string,
};

export default connect(select)(Home);
