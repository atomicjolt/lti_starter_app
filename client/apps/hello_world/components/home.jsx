// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import assets from '../libs/assets';
import Auth from '../../../libs/canvas/components/canvas_authentication';
import Selector from './content_item_select/selector';

const select = state => ({
  canvasAuthRequired: state.settings.canvas_auth_required,
  ltiMessageType: state.settings.lti_message_type,
});

type Props = {
  canvasAuthRequired: boolean,
  ltiMessageType: string,
};

function Home(props :Props) :React.Node {
  const img = assets('./images/atomicjolt.jpg');

  if (props.ltiMessageType === 'ContentItemSelectionRequest') {
    return <Selector />;
  }

  return (
    <div>
      <img src={img} alt="Atomic Jolt Logo" />
      <hr />
      <Auth />
      <hr />
      { props.canvasAuthRequired ? <Auth /> : null }
    </div>
  );

}

export default connect(select)(Home);
