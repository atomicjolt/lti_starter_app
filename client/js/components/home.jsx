import React  from 'react';
import assets from '../libs/assets';

export default function Home() {
  const img = assets('./images/atomicjolt.jpg');

  return (
    <div>
      <img src={img} alt="Atomic Jolt Logo" />
    </div>
  );
}
