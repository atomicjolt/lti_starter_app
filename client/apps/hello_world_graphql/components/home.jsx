import React from 'react';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import { useQuery } from '@apollo/react-hooks';
import { withSettings } from 'atomic-fuel/libs/components/settings';
import { useSelector } from 'react-redux';

import DeepLink from './deep_link';

import assets from '../libs/assets';
import { displayCanvasAuth } from '../../../common/components/common/canvas_auth';

export const GET_WELCOME = gql`
  query getWelcome($name: String!) {
    welcomeMessage(name: $name)
  }
`;

const Home = ({ settings }) => {
  const authRequired = useSelector((state) => state.canvasErrors.canvasReAuthorizationRequired);
  const { loading, error, data } = useQuery(GET_WELCOME, {
    variables: {
      name: 'World',
    },
  });

  if (loading) return 'Loading...';
  if (error) return `Error! ${error.message}`;

  const img = assets('./images/atomicjolt.jpg');

  if (settings.deep_link_settings) {
    return (
      <DeepLink
        deepLinkSettings={settings.deep_link_settings}
      />
    );
  }

  return (
    <div>
      <img src={img} alt="Atomic Jolt Logo" />
      <p>
        {data.welcomeMessage}
      </p>
      <p>
        by
        <a href="http://www.atomicjolt.com">Atomic Jolt</a>
      </p>
      { displayCanvasAuth(settings, authRequired) }
    </div>
  );

};

Home.propTypes = {
  settings: PropTypes.shape({
    deep_link_settings: PropTypes.object,
  }).isRequired,
};

export default withSettings(Home);
