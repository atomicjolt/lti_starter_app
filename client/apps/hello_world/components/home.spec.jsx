import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Home } from './home';

describe('home', () => {
  it('renders the home component', () => {
    const props = {
      canvasRequest: () => {},
      settings: {
        canvas_auth_required: false,
      },
    };
    const result = TestRenderer.create(<Home {...props} />);
    expect(result).toMatchSnapshot();
  });

});
