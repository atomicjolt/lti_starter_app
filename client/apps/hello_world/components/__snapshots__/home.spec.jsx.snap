// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home renders the home component 1`] = `
<div>
  <div>
    <img
      alt="Atomic Jolt Logo"
      src="test-file-stub"
    />
    <hr />
    <h2>
      Courses:
    </h2>
    <ul />
  </div>
  <div>
    <hr />
    <h3>
      Users (LTI Advantage names and roles example):
    </h3>
    <ul
      style={
        Object {
          "textAlign": "left",
        }
      }
    />
    <h3>
      Line items (LTI Advantage line items example):
    </h3>
    <ul
      style={
        Object {
          "textAlign": "left",
        }
      }
    />
    <h3>
      Results (LTI Advantage line item results example):
    </h3>
    <ul
      style={
        Object {
          "textAlign": "left",
        }
      }
    />
  </div>
</div>
`;
